### ISSUE TITLE
Representations::RedirectController raise ActiveSupport::MessageVerifier::InvalidSignature with an invalid :variation_key

### ISSUE THREAD

**Issue by factor4** [rails/rails#41772]
SearchEngineBots often hit ActiveStorage URLs in my Apps with an expired/invalid :variation_key.

### Steps to reproduce
```
git clone https://github.com/factor4/check_activestorage_variantrecord_existence
cd check_activestorage_variantrecord_existence
bundle
rails db:create
rails db:migrate
rails s
```
Visit [http://localhost:3000/](http://localhost:3000/) and add a User with an avatar. On the User#show page ther are two URLs. One with the correct variation_key and one with a invalid one.

### Expected behavior
The first URL response with `302 Found` as expected. The second one should response with a `404 Not Found`.

### Actual behavior
The second one response with `500 Internal Server Error`.

### My quick Fix
I created a branch `fix/activesupport_messageverifier_invalidsignature` in my repo above which fix it. It just check if the VariantRecord exists within the ActiveStorage::SetBlob concern. I'm not sure if that is the right place to fix it, but works for me.

### System configuration
**Rails version**: 6.1.3.1

**Ruby version**: 2.6.6

**Comment by georgeclaghorn** [comment]
Fixed in 657b97e and 7bcbb1c.

**Comment by factor4** [comment]
Thanks for the fix!
